### DISCUSSION TITLE
Allow discuss non github users

### DISCUSSION THREAD

**Discussion by ArtyomCZ** [community/community#2864] [Discussions]
Hi, Please allow people who do not have a github account to discuss. There is great potential to use discussions as support forum and bug report tool for users. It can be used for communication between users and software developers

**Comment by windowsair** [comment]
At least you should give a button to post a new discussion when user are not logged in...

**Reply by becca** [reply]
Tracking this in https://github.com/github/feedback/discussions/29

**Reply by JoseDeFreitas** [reply]
Hello. By this, I guess you mean being able to enable an option for any discussion you want to make it available for even non-registered users? If so, I agree. The option should be disabled by default but if you enable it non-registered users can post comments. This should prevent bad behaviors on the other discussions of the repository. Along with this, there should be also a system to control comments or related. The comments below this one give nice ideas about this 👍.

**Comment by Slatebreaker** [comment]
It would be fantastic as a platform for getting beta feedback from average users who aren't tech-savvy, and as a Q&A, etc etc etc. But requiring them to sign up for a GitHub account can turn them off from using the feature.

**Reply by ArtyomCZ** [reply]
That's right, people usually don't want to register but like to report a bug in the software they use

**Reply by becca** [reply]
Why is signing up for an account a turn off from using the feature, @HazmatDrone?

**Reply by Slatebreaker** [reply]
For myself and other developers, obviously it's not a problem at all; we all have GitHub accounts already.

The problem arises with, let's say, a video game repository. There will be developers, who will then want to share their work with friends, family, random testers online, many of which will have zero coding or git experience.

Signing up for a service is a psychological barrier that can be the difference between one of these such users providing valuable feedback, or just apathetically leaving. With issues, this was kind of okay, as you need a degree of understanding to open an issue in the first place. But with discussions, we have an opportunity to grab feedback from all our users in a much more casual way, and so I personally think that allowing them to post a thread or comment without going through the signup process, or at the very least, *a very simplified one*, would really help developers get more feedback.

Would anyone else agree with me here?

**Reply by ArtyomCZ** [reply]
@HazmatDrone I couldn't have said it better myself.

**Comment by Andre601** [comment]
I for my part would see this as a possible place where someone could spam requests and cause a lot of issues.

By having it require an account does this reduce this potential slightly. I don't say it will be completely gone, but at least would it be reduced compared to when anyone can anonymously create a spam post. And with the current state where discussions don't seem to be delete-able would this create a never ending collection of spam.

**Reply by Slatebreaker** [reply]
It could be a feature that developers can opt into, of course. For massive projects, there is a risk of spam for sure, so the ability to enable/disable it (alongside the ability to delete threads!) would be crucial. But for small or medium repositories, it could still be very beneficial.

**Reply by ArtyomCZ** [reply]
@Andre601 Yes, there should also be a possibility to delete inappropriate posts to avoid spam. There could be a problem with anonymous users, but I don't think that would be unsolvable.

**Reply by WrichikBasu** [reply]
Discussions started by unregistered users can be held for review by repo contributors before being made public. This will greatly reduce spam posts.

**Reply by windowsair** [reply]
> Discussions started by unregistered users can be held for review by repo contributors before being made public. This will greatly reduce spam posts.

This is the correct approach. Through verification code verification and owner review mechanism, we can fully obtain the same cost-effectiveness as registered users

**Comment by extratone** [comment]
Definitely understand many of the above considerations and agree with some of them, but it's worth noting that the OP was simply asking for *the option* for non-registered user posting, generally (unless I'm incorrect.) 

Obviously, requiring registration *by default* makes 110% sense - I would imagine agreeing to some legal/TOU addendum would be required for the admin - but dang... Even the ability to make a single Discussion/thread available for moderator-approved public posts would be absolutely wonderful for me.

Here's why: I use a GitHub repository for my blog (editorially - it began as an experiment and has become my preferred creative existence,) and GitHub Discussions are quickly becoming my favorite forum software in terms of function and aesthetics. I really can't bring myself to like Reddit... But I understand if we're reaching a bit.
